spec_to_code.py

```diff
@@ -155,20 +163,223 @@ def load_language_config() -> LanguageConfigResult:
+                continue
+            if not isinstance(settings, dict):
+                continue
             languages[lang] = LanguageConfig.from_dict(settings)
 
-    return LanguageConfigResult(
-        languages=languages,
-        common_compilation_errors=config.get("common_compilation_errors", []),
-    )
+        return LanguageConfigResult(
+            languages=languages,
+            common_compilation_errors=common_compilation_errors,
+        )
+    except KeyError as e:
+        raise ValueError(
+            f"Missing required configuration key in {config_path}: {e}"
+        ) from e
+    except Exception as e:
+        raise ValueError(
+            f"Error processing configuration from {config_path}: {e}"
+        ) from e
+
+
+# Abstract LLM Interface
+class LLMProvider(ABC):
+    """Abstract interface for LLM providers."""
+
+    def __init__(
+        self, api_key: str, model: str, max_tokens: int = 8192, timeout: float = 60.0
+    ):
+        self.api_key = api_key
+        self.model = model
+        self.max_tokens = max_tokens
+        self.timeout = timeout
+
+    @abstractmethod
+    def call_api(self, prompt: str) -> str:
+        """Call the LLM API with the given prompt."""
+        pass
+
+    @abstractmethod
+    def get_required_env_var(self) -> str:
+        """Return the required environment variable name for API key."""
+        pass
+
+
+class AnthropicProvider(LLMProvider):
+    """Anthropic Claude LLM provider."""
+
+    def __init__(
+        self, api_key: str, model: str = "claude-3-5-sonnet-20241022", **kwargs
+    ):
+        super().__init__(api_key, model, **kwargs)
+        self.client = anthropic.Anthropic(api_key=api_key)
+
+    def call_api(self, prompt: str) -> str:
+        try:
+            response = self.client.messages.create(
+                model=self.model,
+                max_tokens=self.max_tokens,
+                messages=[{"role": "user", "content": prompt}],
+                timeout=self.timeout,
+            )
+
+            if response.content and len(response.content) > 0:
+                text_content = response.content[0]
+                if hasattr(text_content, "text"):
+                    return text_content.text
+                else:
+                    return str(text_content)
+            else:
+                raise ValueError("Unexpected response format from Claude API")
+
+        except anthropic.APIError as e:
+            raise ValueError(f"Anthropic API error: {str(e)}")
+        except Exception as e:
+            raise ValueError(f"Error calling Claude API: {str(e)}")
+
+    def get_required_env_var(self) -> str:
+        return "ANTHROPIC_API_KEY"
+
+
+class OpenAIProvider(LLMProvider):
+    """OpenAI GPT LLM provider."""
+
+    def __init__(self, api_key: str, model: str = "gpt-4o", **kwargs):
+        super().__init__(api_key, model, **kwargs)
+        try:
+            import openai
+
+            self.client = openai.OpenAI(api_key=api_key)
```

Comment:
for all these, just assume the import will work and float to the top level. all this error handling is ugly